### PR TITLE
Added feature to inserte meta-data into template and rewrite of the autobuild tool

### PR DESCRIPTION
**Pull request by noahreinalter**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "teeny-cli",
-    "version": "0.1.0-beta",
+    "version": "0.2.0",
```
Comment (yakkomajuri) [inline package.json]: I'd still call it beta at this stage
```suggestion
    "version": "0.2.0-beta",
```

File: package.json [hunk 2]
```diff
@@ -16,17 +16,17 @@
         "dev:build": "npm run dev && teeny build",
         "dev:develop": "npm run dev && teeny develop",
         "dev:init": "npm run dev && teeny init",
-        "dev": "rm teeny-* || true && npm pack && npm i -g teeny-*.tgz -f",
+        "dev": "sudo npm uninstall -g teeny-cli && (rm teeny-* || true) && npm pack && sudo npm i -g teeny-*.tgz",
```
Comment (yakkomajuri) [inline package.json]: Hmm maybe we can add another `dev:` script?

Uninstalling the global `teeny-cli` feels like too much for this dev script to do - even though I see why you've added it here

File: cli.js [hunk 3]
```diff
@@ -159,3 +204,7 @@ async function safeExecute(func) {
         await func()
     } catch {}
 }
+
+function isNotHiddenFile(src, dest) {
```
Comment (yakkomajuri) [inline cli.js]: why do we take `dest` as an argument here?

File: cli.js [hunk 4]
```diff
@@ -159,3 +204,7 @@ async function safeExecute(func) {
...
+    return !src.match(/.+\/\..*/)
```
Comment (yakkomajuri) [inline cli.js]: Is the idea here to match files like `.gitignore`?

Because if so, it doesn't?

File: LICENSE [hunk 5]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (yakkomajuri) [inline LICENSE:1]: 👍

File: README.md [hunk 6]
```diff
@@ -112,6 +112,12 @@ The last command that Teeny supports is `teeny develop`. This creates an HTTP se
 
 It listens for changes to the files and updates the static files on the fly (naively, by just rebuilding everything each time it detects a change).
 
+## 🤖 Advanced Features
+
+**Replace**
```
Comment (yakkomajuri) [inline README.md:117]: I'd love to have a bit more context into how to use it here

**Review by yakkomajuri — COMMENTED** [review]
Hey @noahreinalter, thanks for this! Replace looks useful. Left a few comments + was wondering if you could write up a bit about the changes in the description e.g. why was "rewrite of the autobuild tool" necessary

Really appreciate the contribution though!

File: cli.js (hunk 4 above)
Comment (noahreinalter) [reply cli.js]: It should match every file or folder beginning with a dot. In my testing it is working as intended but I will look into this

File: cli.js (hunk 3 above)
Comment (noahreinalter) [reply cli.js]: I included it because I didn't know if I needed it and forgot to remove it. It's gone now

**Comment by noahreinalter** [comment]
The rewrite of the autobuild tool was not necessary but hypothetical for very large applications it would be faster. But to be honest the main reason was that you wrote that you are not happy with the current implementation and I had time to change it

File: README.md [hunk 7]
```diff
@@ -110,10 +110,62 @@ Later I did surrender to the `static/` directory approach though, as there may b
...
+
+</body></html>
+```
 
 ## 🔮 Potential future improvements
 
 I want to keep Teeny as tiny as possible. I deliberately put all the code in one file as a reminder to myself that this is supposed to just be a simple tool for me to build simple static blogs quickly.
 
-However, it could use a few "developer experience" upgrades, like an optimized approach to `teeny develop` instead of naively rebuilding everything, as well as some better customization options.
+Add better customization options.
```
Comment (yakkomajuri) [inline README.md]:
```suggestion
However, it could use a few "developer experience" upgrades, as well as some better customization options.
```

**Review by yakkomajuri — APPROVED** [review]
will push this to dev and do some testing when I get a chance - I'll then merge into main

**Comment by yakkomajuri** [comment]
I should probably add some tests at some point too 😅

**Comment by noahreinalter** [comment]
I'm currently working on adding Jest for testing but I'm not sure how much time I have the next days to complete it

**Pull request merged**